api/api_public_transaction_pool.go

```diff
@@ -88,19 +88,27 @@ func (s *PublicTransactionPoolAPI) GetTransactionByBlockHashAndIndex(ctx context
 // GetRawTransactionByBlockNumberAndIndex returns the bytes of the transaction for the given block number and index.
 func (s *PublicTransactionPoolAPI) GetRawTransactionByBlockNumberAndIndex(ctx context.Context, blockNr rpc.BlockNumber, index hexutil.Uint) (hexutil.Bytes, error) {
 	block, err := s.b.BlockByNumber(ctx, blockNr)
-	if block != nil && err == nil {
-		return newRPCRawTransactionFromBlockIndex(block, uint64(index)), nil
+	if err != nil {
+		return nil, err
 	}
-	return nil, err
+	tx := newRPCRawTransactionFromBlockIndex(block, uint64(index))
+	if tx == nil {
+		return nil, fmt.Errorf("the transaction does not exist (BlockNum: %d, Index:%d)", blockNr, index)
+	}
+	return tx, nil
 }
 
 // GetRawTransactionByBlockHashAndIndex returns the bytes of the transaction for the given block hash and index.
 func (s *PublicTransactionPoolAPI) GetRawTransactionByBlockHashAndIndex(ctx context.Context, blockHash common.Hash, index hexutil.Uint) (hexutil.Bytes, error) {
 	block, err := s.b.BlockByHash(ctx, blockHash)
-	if block != nil && err == nil {
-		return newRPCRawTransactionFromBlockIndex(block, uint64(index)), nil
+	if err != nil {
+		return nil, err
 	}
-	return nil, err
+	tx := newRPCRawTransactionFromBlockIndex(block, uint64(index))
+	if tx == nil {
+		return nil, fmt.Errorf("the transaction does not exist (BlockHash: %s, Index:%s)", blockHash.String(), index)
```

Comment:
What about telling the number of txs? It would be helpful if it tells the total index of a block. If you think it's not necessary, please ignore my comment since it's just a suggestion.
`the transaction does not exist (Blockhash: %s, Index: %s, number of txs in the block: %d)`

Reply:
resolved by [205ceae](https://github.com/klaytn/klaytn/pull/1874/commits/205ceaec7a982d541336adc5385fa3c97e845f68)

return nil intead of error